isolate eval call so it can't muck with local variables

nerdrew opened the pull request:
- update rubocop, and fix all rubocop failures
- Refactor specs

doctown commented on lib/debug_socket.rb:
```diff
@@ -79,7 +83,14 @@ def self.backtrace
+  rescue Exception => e # rubocop:disable Lint/RescueException
+    logger&.error "debug-socket-error=callback unsuccessful: #{e.inspect} for #{input.inspect} path=#{@path} backtrace=#{e.backtrace.inspect}"
+  end
+
```
Might be worth adding a commend explaining why this is being introduced or what we're trying to avoid.

nerdrew replied:
Good call. I think I'm going to move it into it's own empty module in fact.

doctown reviewed (approved):
Thanks for working on this.

nerdrew commented on lib/debug_socket.rb:
```diff
@@ -32,22 +32,27 @@ def self.start(path, &block)
+        socket&.close
       end
+    rescue Exception => e # rubocop:disable Lint/RescueException
+      logger&.error { "debug-socket-error=#{e.inspect} DebugSocket is broken now path=#{@path} backtrace=#{e.backtrace.inspect}" }
```
There is no other socket. This message means the debug socket server is broken and the process would need to be restarted. This is a fatal (for the debug socket at least) error.

nerdrew replied:
Added.

nerdrew replied:
I don't know if we have ever seen this in practice. BUT I noticed it when testing. When I typo-ed a line, there were a LOT of logs spewing the error because we had a `rescue Exception` inside a loop. I've now changed it to have a `rescue Exception` on the `eval` above (we don't want user input to disrupt anything). But we shouldn't be rescuing `SyntaxError`s in this file itself.

Now, we only allow 20 errors that aren't cause by the `eval` itself, e.g. a socket error can happen if the client writes to the socket and then closes really quickly before it has time to respond. I _think_ that is ok. I don't know how to distinguish "there's something wrong with the socket and we shouldn't infinite loop spewing errors" vs "the client did something wrong and we should rescue the error and keep going". I think it's better to fail VERY conservatively for a debug tool. I never want `DebugSocket` to interrupt or negatively impact a running process (of course you can do that yourself by sending `exit` as your command, but don't do that).

nerdrew commented on lib/debug_socket.rb:
```diff
@@ -5,6 +5,29 @@
+  module Commands
+    # When running `eval`, we don't want the input to overwrite local variables etc. `eval` runs in the current scope,
+    # so we have an empty scope here that runs in a module that only has other shortcut commands the client might want
+    # to run.
```
@doctown comment as requested.

doctown commented:
Not sure if its worth updating the CHANGELOG with this change. Most is encapsulated in the change I mentioned.

nerdrew committed:
isolate eval call so it can't muck with local variables

- update rubocop, and fix all rubocop failures
- Refactor specs
- add github actions CI
- remove travis CI

nerdrew commented:
> Not sure if its worth updating the CHANGELOG with this change. Most is encapsulated in the change I mentioned.

Ah, I should add a changelog entry. This is a big change (in terms of lines of code changed).
